components/brave_wallet/browser/brave_wallet_utils_unittest.cc

```diff
@@ -766,12 +766,21 @@ TEST(BraveWalletUtilsUnitTest, GetInfuraSubdomainForKnownChainId) {
   prefs.registry()->RegisterDictionaryPref(kBraveWalletCustomNetworks);
   prefs.registry()->RegisterBooleanPref(kSupportEip1559OnLocalhostChain, false);
 
+  base::flat_set<std::string> infura_chains = {
+      brave_wallet::mojom::kMainnetChainId,
+      brave_wallet::mojom::kPolygonMainnetChainId,
+      brave_wallet::mojom::kRinkebyChainId,
+      brave_wallet::mojom::kRopstenChainId,
+      brave_wallet::mojom::kGoerliChainId,
+      brave_wallet::mojom::kKovanChainId};
+
   std::vector<mojom::NetworkInfoPtr> known_chains;
   GetAllKnownEthChains(&prefs, &known_chains);
   for (const auto& chain : known_chains) {
-    auto subdomain = GetInfuraSubdomainForKnownChainId(chain->chain_id);
-    bool expected = (chain->chain_id == brave_wallet::mojom::kLocalhostChainId);
-    ASSERT_EQ(subdomain.empty(), expected);
+    auto network_url =
+        GetNetworkURL(&prefs, chain->chain_id, mojom::CoinType::ETH);
```

Comment:
How about we rename the test case and test GetInfuraURLForKnownChainId instead, call `GetInfuraURLForKnownChainId` for known eth chains and make sure they have exact expected hostnames or empty strings for those public RPC endpoints.